### ISSUE TITLE
Playground counts each non-BMP code point as two code points

### ISSUE THREAD

**Issue by dscorbett** [astral-sh/ruff#16266]
### Description

The playground counts each non-BMP code point as two code points when applying fixes, leaving characters there that should have been replaced as part of the fix. [Here is an example.](https://play.ruff.rs/8aef5a51-fa9e-4d87-9634-537659f3c65a) The input is:
```python
dict(𠤪=1)
```
Applying the quick fix for [`unnecessary-collection-call` (C408)](https://docs.astral.sh/ruff/rules/unnecessary-collection-call/) incorrectly produces:
```python
{"𠤪": 1})
```
On the command line, though, Ruff 0.9.6 correctly fixes it to:
```python
{"𠤪": 1}
```
